### DISCUSSION TITLE
can  you download older version or branch

### DISCUSSION THREAD

**Discussion by krywenko** [Koenkk/zigbee2mqtt#10420] [General]
hi there I was wonder  can  you download an older version  it tried git fetch --tags   but there does not seam to be  any older version in the branches

or perhaps i am just doing it wrong   I want to install zigbee2mqtt on an openwrt router platform   older versions worked fine  as openwrt only support node 8.16 .. and it only relatively recently that you dropped support for node +8.10

**Comment by mausch** [comment]
Don't know about running z2m on openwrt but all releases are tagged, check out https://github.com/Koenkk/zigbee2mqtt/tags

**Reply by krywenko** [reply]
thank you I will give them a try it. no problem running Z2M on openwrt   being doing it for about 3 years now..here a post I made about how too - https://forum.openwrt.org/t/howto-setup-zigbee2mqtt-on-openwrt/  --  but in the newer version openwrt they support node v8.16  and I do not have to compile it anymore  .   but the newer Z2M wants +node v10 ..   so it faster to install an older version of z2m that supports node v8.16 then to compile  node v10 for openwrt - it come down to 20 minutes for z2m to install versus   +24 hrs for openwrt to compile node v10  for it  and then another 20 minutes to install z2m ---  by  chance do you  know the last stable version that supported node v8.16

**Reply by mausch** [reply]
Hi, you can see that by doing a git blame on the Dockerfile.
Starting with https://github.com/Koenkk/zigbee2mqtt/blame/master/docker/Dockerfile navigate on the history of the first line of this file.
Eventually you'll see that [nodejs isn't pinned in the dockerfile any more](https://github.com/Koenkk/zigbee2mqtt/commit/9a21e9d956485cbdefcc993b8e8d29a047c06666).
From that point you can keep going back in time through the `.travis.yml` file ( https://github.com/Koenkk/zigbee2mqtt/blame/9a21e9d956485cbdefcc993b8e8d29a047c06666/.travis.yml )
This commit shows when node was upgraded from v8 to v10 in Travis, on Nov 7, 2018 
: https://github.com/Koenkk/zigbee2mqtt/commit/fc664bdc6809a6f93c18d75984d4f9cc99d1d5fd , though it's odd that the dockerfiles on that commit pin node to 8.11 :man_shrugging: 
Then look for the last release before that date, which was https://github.com/Koenkk/zigbee2mqtt/releases/tag/0.1.8

Having said all that, you _might_ be able to build later releases with node 8 too but this should give you a rough starting point.

**Reply by krywenko** [reply]
thank you for the reply -- well I just bit the bullet and complied the newer  node and npm  and other support file for zigbee2mqtt  for openwrt   took  like I said about  24 hrs ..  support went abit farther then 2018   my  first compile was 2019  and I am pretty sure node >= v8.11  still worked in 2021  either v 1.18 or 1.19  was the last time i tried
